### PR TITLE
Support authentication for trace configuration

### LINKED ISSUE #123531: Support endpoint authentication for K8s trace configuration
### What would you like to be added?

So far, grpc format endpoint is supported for K8s trace configuration.
If user wants to use an authentication for trace, it is not feasible.
I want to support grpc authentication for trace configuration.

### Why is this needed?

Only grpc without authentication for trace config is not enough, we need to support
grpc authentication for trace

### PR DESCRIPTION
**Pull request by CasperLiu**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR introduces authentication for K8s trace configuration, enabling K8s trace spans to be emitted to trace storage backends that require essential authentication (such as an OpenTelemetry Collector with an auth plugin or a cloud provider's trace storage product).

Up until now, K8s trace configuration has supported options for endpoint and sampling rate per million.

In certain scenarios, K8s can emit spans directly to a provider’s trace storage product, eliminating the need for an OpenTelemetry Collector, which would incur additional resource costs and operational complexity. Therefore, authentication is necessary for K8s in this context.

Furthermore, in scenarios where an OpenTelemetry Collector with authentication is used to enhance security, K8s also requires authentication.

In summary, introducing authentication for trace configuration is essential.

#### Which issue(s) this PR fixes:

Fixes #123531

#### Special notes for your reviewer:
CC @dashpole 
Could you please help review this? Thanks!

#### Does this PR introduce a user-facing change?

```release-note
Introduces authentication for K8s trace configuration, enabling K8s trace spans to be emitted to trace storage backends that require essential authentication (such as an OpenTelemetry Collector with an auth plugin or a cloud provider's trace storage product).
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Comment by dashpole** [comment]
/hold
See https://github.com/kubernetes/kubernetes/issues/123531#issuecomment-1978986352

**Comment by bart0sh** [comment]
@CasperLiu How are you going proceed with this, considering above @dashpole's comment?

**Comment by CasperLiu** [comment]
> @CasperLiu How are you going proceed with this, considering above @dashpole's comment?

Hello Ed,
I have verified solution from David, it works well expected. Great~
Thanks!

Casper
